File: 파이썬/완주하지못한선수.py
def solution(participant, completion):
    NOT_COMPLE = 1
    table = {}
    for p in participant:
        table[p] = table.get(p, 0) + 1
    for c in completion:
        table[c] -= 1
    return [k for k, v in table.items() if v == NOT_COMPLE][0]

File: 자바스크립트/완주하지못한선수.js
function solution(participant, completion) {
    const [NOT_COMPL, NAME, IS_COMPL] = [1, 0, 1]
    
    let table = {}
    for (const p of participant){
        table[p] = (table[p] || 0) + 1
    }
    for (const c of completion){
        table[c] -= 1
    }
    
    const compleStates = Object.entries(table)
    return compleStates.find(person => person[IS_COMPL] === NOT_COMPL)[NAME]
}